src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -81,11 +82,17 @@
     private final String emailFieldName;
     private final String scopes;
     private final boolean disableSslVerification;
+    private final boolean logoutFromOpenidProvider;
+    private final String endSessionUrl;
+    private final String postLogoutRedirectUrl;
+
+    private IdTokenResponse idTokenResponse;
```

Comment:
Hmm I think this is shared among users and IdTokenResponse source indicates that it's not thread safe.